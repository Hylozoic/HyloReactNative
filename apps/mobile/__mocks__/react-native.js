import React, {useState} from 'react'
import * as ReactNative from 'react-native'

export const Switch = props => {
  const [value, setValue] = useState(props.value)
  return (
    <ReactNative.TouchableOpacity
      onPress={() => {
        props.onValueChange(!value)
        setValue(!value)
      }}
      testID={props.testID}>
      <ReactNative.Text>{value?.toString()}</ReactNative.Text>
    </ReactNative.TouchableOpacity>
  )
}

newRN = Object.defineProperty(ReactNative, 'Switch', {
  get: function() {
    return Switch
  }
})

module.exports = newRN
