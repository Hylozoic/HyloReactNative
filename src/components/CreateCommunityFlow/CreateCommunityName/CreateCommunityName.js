import React from 'react'
import {
  Text,
  View,
  TextInput
} from 'react-native'
import { HeaderBackButton } from 'react-navigation'
import Button from '../../Button'
import KeyboardFriendlyView from '../../KeyboardFriendlyView'
import ErrorBubble from '../../ErrorBubble'
import header from 'util/header'
import styles from '../CreateCommunityFlow.styles'
import { caribbeanGreen, white, white60onCaribbeanGreen } from 'style/colors'

export default class CreateCommunityName extends React.Component {
  static navigationOptions = ({ navigation }) => {
    return header(navigation, {
      title: 'Step 1/3',
      options: {
        headerLeft: <HeaderBackButton onPress={() => navigation.goBack()} tintColor={white60onCaribbeanGreen} />,
        headerBackTitle: null,
        headerBackgroundColor: white
      },
      customStyles: {
        header: {
          backgroundColor: caribbeanGreen,
          paddingHorizontal: 10
        },
        title: {
          color: white
        }
      }
    })
  }

  constructor (props) {
    super(props)
    this.state = {
      communityName: this.props.communityName
    }
  }

  setInput (key, value) {
    this.setState({
      ...this.state,
      [key]: value
    })
  }

  clearErrors = () => {
    this.setState({
      ...this.state,
      error: null
    })
  }

  checkAndSubmit = () => {
    const { communityName } = this.state

    this.clearErrors()
    if (!communityName || communityName.length === 0) {
      this.setState({
        ...this.state,
        error: 'Please enter a community name'
      })
      return
    }

    this.setState({
      ...this.state,
      error: null
    })
    this.props.saveCommunityName(communityName)
    this.props.goToCreateCommunityUrl()
  }

  render () {
    const { error, communityName } = this.state
    return <KeyboardFriendlyView style={styles.container}>
      <Text style={styles.header}>Lets get started!</Text>
      <Text style={styles.description}>All good things start somewhere! Lets kick things off with a catchy name for your community.</Text>
      <View style={styles.textInputContainer}>
        <Text style={styles.textInputLabel}>Whats the name of your community?</Text>
        <TextInput
          style={styles.textInput}
          onChangeText={communityName => this.setInput('communityName', communityName)}
          returnKeyType='next'
          autoCapitalize='none'
          value={communityName}
          autoCorrect={false}
          underlineColorAndroid={styles.androidInvisibleUnderline} />
      </View>
      {error && <ErrorBubble styles={null} text={'Please enter a community name'} />}

      <Button text='Continue' onPress={this.checkAndSubmit} style={styles.button} disabled={false} />
    </KeyboardFriendlyView>
  }
}
