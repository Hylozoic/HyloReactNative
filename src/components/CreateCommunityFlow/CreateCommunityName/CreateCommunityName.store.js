export const MODULE_NAME = 'CreateCommunityName'
export const SAVE_COMMUNITY_NAME = `${MODULE_NAME}/SAVE_COMMUNITY_NAME`

export default function reducer (state = {}, action) {
  const { type, payload } = action
  switch (type) {
    case SAVE_COMMUNITY_NAME:
      return {
        ...state,
        communityName: payload
      }
  }
  return state
}

export function saveCommunityName (name) {
  return {
    type: SAVE_COMMUNITY_NAME,
    payload: name
  }
}
