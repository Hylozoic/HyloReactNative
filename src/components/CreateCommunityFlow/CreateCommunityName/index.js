import component from './CreateCommunityName'
export default component
