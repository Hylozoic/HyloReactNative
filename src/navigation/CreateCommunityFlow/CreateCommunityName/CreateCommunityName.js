import React from 'react'
import {
  Text,
  View,
  TextInput
} from 'react-native'
import SafeAreaView from 'react-native-safe-area-view'
import Button from 'components/Button'
import KeyboardFriendlyView from 'navigation/KeyboardFriendlyView'
import ErrorBubble from 'components/ErrorBubble'
import styles from '../CreateCommunityFlow.styles'
import createCommunityHeader from '../util/createCommunityHeader'

export default class CreateCommunityName extends React.Component {
  static navigationOptions = ({ navigation, route }) => {
    const title = 'STEP 1/3'
    return createCommunityHeader(title, navigation)
  }

  constructor (props) {
    super(props)
    this.state = {
      communityName: this.props.communityName
    }
  }

  setInput (key, value) {
    this.setState({
      ...this.state,
      [key]: value
    })
  }

  clearErrors = () => {
    this.setState({
      ...this.state,
      error: null
    })
  }

  checkAndSubmit = () => {
    const { communityName } = this.state

    this.clearErrors()
    if (!communityName || communityName.length === 0) {
      this.setState({
        ...this.state,
        error: 'Please enter a community name'
      })
      return
    }
    this.props.saveCommunityName(communityName)
    this.props.goToCreateCommunityUrl()
  }

  render () {
    const { error, communityName } = this.state
    return (
      <SafeAreaView style={styles.container}>
        <KeyboardFriendlyView>
          <View style={styles.header}>
            <Text style={styles.heading}>Let's get started!</Text>
            <Text style={styles.description}>All good things start somewhere! Let's kick things off with a catchy name for your community.</Text>
          </View>
          <View style={styles.content}>
            <View style={styles.textInputContainer}>
              <Text style={styles.textInputLabel}>What's the name of your community?</Text>
              <TextInput
                style={styles.textInput}
                onChangeText={communityName => this.setInput('communityName', communityName)}
                returnKeyType='next'
                autoCapitalize='none'
                value={communityName}
                autoCorrect={false}
                underlineColorAndroid={styles.androidInvisibleUnderline}
              />
            </View>
            {error && <View style={styles.errorBubble}><ErrorBubble text={error} topArrow /></View>}
          </View>
          <View style={styles.footer}>
            <Button text='Continue' onPress={this.checkAndSubmit} style={styles.button} />
          </View>
        </KeyboardFriendlyView>
      </SafeAreaView>
    )
  }
}
