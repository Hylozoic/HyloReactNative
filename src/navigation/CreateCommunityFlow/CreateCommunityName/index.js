import component from './CreateCommunityName'
import connector from './CreateCommunityName.connector'
export default connector(component)
