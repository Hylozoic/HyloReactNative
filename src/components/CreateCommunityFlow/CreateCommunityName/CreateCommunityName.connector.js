import { connect } from 'react-redux'
import { saveCommunityName } from '../CreateCommunityFlow.store'

export const mapDispatchToProps = {
  saveCommunityName
}

export function mergeProps (stateProps, dispatchProps, ownProps) {
  const { navigation } = ownProps
  return {
    ...stateProps,
    ...dispatchProps,
    ...ownProps,
    goToCreateCommunityUrl: () => {
      navigation.navigate('CreateCommunityUrl')
    }
  }
}

export default connect(null, mapDispatchToProps, mergeProps)
