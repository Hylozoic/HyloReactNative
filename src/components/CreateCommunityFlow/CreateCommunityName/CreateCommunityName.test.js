import 'react-native'
import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import ReactTestRenderer from 'react-test-renderer'
import CreateCommunityName from './CreateCommunityName'

jest.mock('../../KeyboardFriendlyView', () => 'KeyboardFriendlyView')

it('matches last snapshot', () => {
  const renderer = new ReactShallowRenderer()
  renderer.render(<CreateCommunityName />)
  const actual = renderer.getRenderOutput()
  expect(actual).toMatchSnapshot()
})

it('displays an error message without a community name with checkAndSubmit', () => {
  const renderer = ReactTestRenderer.create(<CreateCommunityName />)
  const instance = renderer.getInstance()
  const communityName = ''
  instance.setState({communityName})
  instance.checkAndSubmit()
  expect(renderer.toJSON()).toMatchSnapshot()
})

it('stores the community name and redirects with a community name with checkAndSubmit', () => {
  const props = {
    saveCommunityName: jest.fn(),
    goToCreateCommunityUrl: jest.fn()
  }
  const communityName = 'communityName'
  const renderer = ReactTestRenderer.create(<CreateCommunityName {...props} />)
  const instance = renderer.getInstance()
  instance.setState({communityName})
  instance.checkAndSubmit()
  expect(props.saveCommunityName).toHaveBeenCalledWith(communityName)
  expect(props.goToCreateCommunityUrl).toHaveBeenCalled()
})

it('clears error from state with clearErrors', () => {
  const renderer = ReactTestRenderer.create(<CreateCommunityName />)
  const instance = renderer.getInstance()
  const error = 'some error'
  instance.setState({error})
  instance.clearErrors()
  expect(instance.state.error).toBeNull()
})

it('updates state with setInput', () => {
  const renderer = ReactTestRenderer.create(<CreateCommunityName />)
  const instance = renderer.getInstance()
  const key = 'key'
  const value = 'value'
  instance.setInput(key, value)
  instance.clearErrors()
  expect(instance.state[key]).toEqual(value)
})

it('sets the initial state with the initial communityName prop value', () => {
  const communityName = 'communityName'
  const props = {
    communityName
  }
  const renderer = ReactTestRenderer.create(<CreateCommunityName {...props} />)
  const instance = renderer.getInstance()
  expect(instance.state.communityName).toEqual(communityName)
})

it('has navigationOptions', () => {
  const navigation = jest.fn()
  expect(CreateCommunityName.navigationOptions(navigation)).toMatchSnapshot()
})
