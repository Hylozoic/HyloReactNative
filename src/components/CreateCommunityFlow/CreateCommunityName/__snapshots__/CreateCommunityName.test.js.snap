// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays an error message without a community name with checkAndSubmit 1`] = `
<RCTScrollView>
  <View>
    <KeyboardFriendlyView>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Let's get started!
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        All good things start somewhere! Let's kick things off with a catchy name for your community.
      </Text>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          What's the name of your community?
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          returnKeyType="next"
          underlineColorAndroid="rgba(0,0,0,0)"
          value=""
        />
      </View>
      <View>
        <View>
          <View />
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Please enter a community name
            </Text>
          </View>
        </View>
      </View>
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View>
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Continue
              </Text>
            </View>
          </View>
        </View>
      </View>
    </KeyboardFriendlyView>
  </View>
</RCTScrollView>
`;

exports[`has navigationOptions 1`] = `
Object {
  "color": "#FFFFFF",
  "fontSize": 10,
  "headerBackgroundColor": "#0DC39F",
  "headerLeft": <HeaderBackButton
    onPress={[Function]}
    pressColorAndroid="rgba(0, 0, 0, .32)"
    tintColor="#9ee7d9"
    truncatedTitle="Back"
/>,
  "headerStyle": Object {
    "backgroundColor": "#0DC39F",
    "borderBottomWidth": 0,
    "elevation": 0,
    "shadowOffset": Object {
      "height": 0,
    },
    "shadowRadius": 0,
  },
  "title": <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
>
    STEP 1/3
</Text>,
}
`;

exports[`matches last snapshot 1`] = `
<ScrollView>
  <KeyboardFriendlyView>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Let's get started!
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      All good things start somewhere! Let's kick things off with a catchy name for your community.
    </Text>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        What's the name of your community?
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        returnKeyType="next"
        underlineColorAndroid="rgba(0,0,0,0)"
        value={undefined}
      />
    </View>
    <Button
      onPress={[Function]}
      text="Continue"
    />
  </KeyboardFriendlyView>
</ScrollView>
`;
